torch/csrc/jit/import.cpp

```diff
@@ -263,8 +263,25 @@ script::Module load(
     std::unique_ptr<ReadAdapterInterface> rai,
     c10::optional<c10::Device> device,
     script::ExtraFilesMap& extra_files) {
+  // Verify that we're loading a zip archive and not a torch.save pickle archive
+  // (marked by the 0x80 0x02 bytes at the start)
+  uint16_t first_short = 0;
+  rai->read(
+      /*pos=*/0,
+      /*buf=*/reinterpret_cast<void*>(&first_short),
+      /*n=*/2,
+      /*what=*/"checking archive");
+  if (first_short == 0x8002) {
```

Comment:
It would be more robust to do a string comparison here, otherwise it is endian flipped. Not that important, but an easy change.